src/metpy/plots/declarative.py

```diff
@@ -1410,14 +1409,21 @@ def name(self):
     @property
     def obsdata(self):
         """Return the internal cached data."""
-        time_vars = ['valid', 'time', 'valid_time']
-        stn_vars = ['station', 'stn']
+        time_vars = ['valid', 'time', 'valid_time', 'date_time', 'date']
+        stn_vars = ['station', 'stn', 'station_id', 'stid']
         if getattr(self, '_obsdata', None) is None:
-            for dim_name in list(self.data):
-                if dim_name in time_vars:
-                    dim_time = dim_name
-                elif dim_name in stn_vars:
-                    dim_stn = dim_name
+            dim_times = [time_var for time_var in time_vars if time_var in list(self.data)]
```

Comment:
is this supposed to be here then?

Reply:
Somehow didn't see your comment above. It should disappear when I merge #1259 and you rebase on it.